rali/rali/source/rali_api_meta_data.cpp

```diff
@@ -104,25 +104,36 @@ RALI_API_CALL raliCreateTextFileBasedLabelReader(RaliContext p_context, const ch
 
 }
 
-
 void
-RALI_API_CALL raliGetImageName(RaliContext p_context,  char* buf, unsigned image_idx)
+RALI_API_CALL raliGetImageName(RaliContext p_context,  char* buf)
 {
     auto context = static_cast<Context*>(p_context);
     auto meta_data = context->master_graph->meta_data();
-    if(image_idx >= meta_data.first.size())
-        THROW("Image idx is out of batch size range")
-    memcpy(buf, meta_data.first[image_idx].c_str(), meta_data.first[image_idx].size());
+    size_t meta_data_batch_size = meta_data.first.size();
+    if(context->user_batch_size() != meta_data_batch_size)
+        THROW("meta data batch size is wrong " + TOSTR(meta_data_batch_size) + " != "+ TOSTR(context->user_batch_size() ))
+    for(unsigned int i = 0; i < meta_data_batch_size; i++)
+    {
+        memcpy(buf, meta_data.first[i].c_str(), meta_data.first[i].size());
+        buf += meta_data.first[i].size() * sizeof(char);
+    }
 }
 
 unsigned
-RALI_API_CALL raliGetImageNameLen(RaliContext p_context, unsigned image_idx)
+RALI_API_CALL raliGetImageNameLen(RaliContext p_context, int* buf)
 {
+    unsigned size = 0;
```

Comment:
Codacy found an issue: [The scope of the variable 'size' can be reduced.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=6302037)